api/src/main/java/org/openmrs/module/registrationcore/api/impl/PatientActionListener.java

```diff
@@ -1,21 +1,31 @@
 package org.openmrs.module.registrationcore.api.impl;
 
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.openmrs.OpenmrsObject;
 import org.openmrs.Patient;
 import org.openmrs.api.APIException;
 import org.openmrs.api.PatientService;
 import org.openmrs.api.context.Daemon;
-import org.openmrs.event.EventListener;
+import org.openmrs.event.Event;
+import org.openmrs.event.SubscribableEventListener;
 import org.openmrs.module.DaemonToken;
-import org.openmrs.module.registrationcore.RegistrationCoreConstants;
 
 import javax.jms.JMSException;
 import javax.jms.MapMessage;
 import javax.jms.Message;
+import org.openmrs.module.registrationcore.api.errorhandling.SendingPatientToMpiParameters;
 
 /**
- * Abstract class for event listening.
+ * Abstract class for subscribable event listening.
  */
-public abstract class PatientActionListener implements EventListener {
+public abstract class PatientActionListener implements SubscribableEventListener {
+    protected final Log log = LogFactory.getLog(this.getClass());
```

Comment:
Is there any reason why you did not make the above private?

Reply:
No reason, I was just copying from other examples, but since it is not used in the subclasses, it can be made private. Changed to "private final"